yt_dlp/extractor/vidyard.py

```diff
@@ -0,0 +1,256 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    extract_attributes,
+    int_or_none,
+    traverse_obj,
+)
+
+
+class VidyardBaseIE(InfoExtractor):
+
+    def _get_formats_and_subtitles(self, video_source, video_id):
+        video_source = video_source or {}
+        formats, subtitles = [], {}
+        for key, value in video_source.items():
```

Comment:
Safer:
```suggestion
        formats, subtitles = [], {}
        for key, value in traverse_obj(video_source, ({dict.items}, lambda _, v: v[1][0])):
```
Now we don't care if video_source is not a `dict`, and any `value` is known to be a list.

Also, `key`, `value` are a bit generic. Maybe `src_type`, `srcs`, or similar?

Reply:
Applied in 4f07c29a31939c0a0a08526a943f1d0d221b24e6